src/Persister/EntityPersister.php

```diff
@@ -99,6 +106,23 @@ public function getUpdateQuery($object)
         return Statement::create($query, ['id' => $id, 'props' => $propertyValues]);
     }
 
+    /**
+     * Refreshes a managed entity.
+     *
+     * @param int  $id
+     * @param object $entity The entity to refresh.
+     *
+     * @return void
+     */
+    public function refresh($id, $entity)
```

Comment:
Maybe check that the class of the entity given is the same as `$this->classMetadata->getClassName()`